Switch to PyPI trusted publishing

agriyakhetarpal opened the pull request:
Closes #193; the environment name will have to be configured in https://github.com/pyodide/micropip/settings/ and the trusted publishing on PyPI (release workflow name, and GitHub environment name) will have to be configured in https://pypi.org/manage/project/micropip/settings/.

I don't have access to either of these pages, so I'll tag you, @hoodmane or @ryanking13, so that you can do this.

ryanking13 reviewed (approved):
Thanks @agriyakhetarpal!

I added the environment in the GitHub setting page, and I also updated trusted publishing setting in PyPI.

ryanking13 commented:
Also, please let me know your PyPI username, I'll add you to the Pyodide projects.

agriyakhetarpal commented:
> Thanks @agriyakhetarpal!
> 
> I added the environment in the GitHub setting page, and I also updated trusted publishing setting in PyPI.

Thanks, @ryanking13! Please delete the `PYPI_API_TOKEN` secret as well, just in case you might not have done so already.

> Also, please let me know your PyPI username, I'll add you to the Pyodide projects.

Thanks! My PyPI username is `agriyakhetarpal`. Here is a link to my profile: https://pypi.org/user/agriyakhetarpal/

ryanking13 commented:
Awesome. Thanks @rth!
